lib/internal/errors.js

```diff
@@ -1450,8 +1450,12 @@ E('ERR_MISSING_ARGS',
     return `${msg} must be specified`;
   }, TypeError);
 E('ERR_MISSING_OPTION', '%s is required', TypeError);
-E('ERR_MODULE_NOT_FOUND', (path, base, type = 'package') => {
-  return `Cannot find ${type} '${path}' imported from ${base}`;
+E('ERR_MODULE_NOT_FOUND', function(path, base, exactUrl) {
+  if (exactUrl) {
+    this.url = exactUrl;
```

Comment:
```suggestion
    setOwnProperty(this, 'url', exactUrl);
```

Reply:
Again, I'm not clear on why this is needed?

Reply:
Ah right, the prototype setter bug.